mlx_lm/generate.py

```diff
@@ -1036,7 +1106,9 @@ def _next(self):
             else:
                 finish_reason = None
                 keep_idx.append(e)
-            responses.append(self.Response(uid, t, logprobs[e], finish_reason))
+            if finish_reason is not None:
+                cache = batch.extract_cache(e)
+            responses.append(self.Response(uid, t, logprobs[e], finish_reason, cache))
```

Comment:
Similar comment here. If we don't need to preserve the cache we should avoid holding onto it. 

For e.g. doing an eval with thousands of prompts this will take up a lot of RAM.